### PR TITLE
119 mainenance cast

### PR DESCRIPTION
**Pull request by tdivis**

### CODE REVIEW

File: eidas_node/models.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def digest(self, hash_algorithm: str, secret: str) -> bytes:
         :raise ValidationError: If token data are invalid.
         """
         self.validate()
-        data = '|'.join((self.id, self.issuer, create_eidas_timestamp(self.created), secret))
+        data = '|'.join((str(self.id), str(self.issuer), create_eidas_timestamp(cast(datetime, self.created)), secret))
```
[the remark this answers is not in the document]
Comment (tdivis) [reply eidas_node/models.py]: OK, rewrote to `assert XXX is not None` here and in several other instances, so it's consistent.

**Comment by tdivis** [comment]
I have added test compatibility with libxmlsec>1.2.34, @tpazderka please have a look.

File: eidas_node/tests/test_xml.py [hunk 2]
```diff
@@ -228,7 +236,11 @@ def test_sign_xml_node_multiple(self, uuid_mock):
         sign_xml_node(assertion, position=0, **SIGNATURE_OPTIONS)
         sign_xml_node(root, position=0, **SIGNATURE_OPTIONS)
 
-        with cast(TextIO, (DATA_DIR / 'signed_response_and_assertion.xml').open('r')) as f:
+        if LIBXMLSEC_VERSION < (1, 2, 35):  # pragma: no cover
+            filename = 'signed_response_and_assertion.xml'
+        else:  # pragma: no cover
+            filename = 'signed_response_and_assertion.xml_1.2.35.xml'
```
Comment (tpazderka) [inline eidas_node/tests/test_xml.py:242]: The file seems to be missing.

File: eidas_node/tests/test_xml.py [hunk 3]
```diff
@@ -206,7 +206,11 @@ def test_sign_xml_node_without_id(self, uuid_mock):
         SubElement(assertion, Q_NAMES['saml2:Issuer']).text = 'Test Issuer'
         sign_xml_node(root, position=0, **SIGNATURE_OPTIONS)
 
-        with cast(TextIO, (DATA_DIR / 'signed_response.xml').open('r')) as f:
+        if LIBXMLSEC_VERSION < (1, 2, 35):  # pragma: no cover
+            filename = 'signed_response.xml'
+        else:  # pragma: no cover
+            filename = 'signed_response_1.2.35.xml'
```
Comment (tpazderka) [inline eidas_node/tests/test_xml.py:212]: This file seems to be missing.

File: .github/workflows/python-test.yml [hunk 4]
```diff
@@ -5,14 +5,15 @@ on:
...
+    - cron: '30 5 * * *'
 jobs:
   build:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-22.04
     strategy:
       max-parallel: 4
       matrix:
-        python: [3.5, 3.6, 3.7, 3.8, 3.9]
+        python: [3.8, 3.9]
```
Comment (tpazderka) [inline .github/workflows/python-test.yml:16]: We are claiming support for python 3.5-3.7 as well, but we are not testing it. 
Out of those, 3.7 is still supported version, so we shouldn't drop that one.
I would add another job that runs the older python version(s)...

**Review by tpazderka — CHANGES_REQUESTED** [review]

**Review by tpazderka — APPROVED** [review]

**Pull request merged**